orderer/consensus/follower/block_puller.go

```diff
@@ -49,6 +50,20 @@ func BlockPullerFromJoinBlock(
 	}
 
 	verifyBlockSequence := func(blocks []*common.Block, _ string) error {
+		if len(blocks) == 0 {
+			return errors.New("buffer is empty")
+		}
+		if blocks[0] == nil {
+			return errors.New("first block is nil")
+		}
+		if blocks[0].Header == nil {
+			return errors.New("first block header is nil")
+		}
+		if blocks[0].Header.Number == 0 {
+			blocksAfterGenesis := blocks[1:]
```

Comment:
why are you doing this? Is it the signature? If it is, then I think there are other ways of not verifying the signature of block 0, but what you're doing here also forces not to validate the hash pointer of block 1, no?

Reply:
Yes, the only reason I am doing this is to avoid verifying the (missing) signature on block 0. 
You are correct - there is a problem here, we need to verify the data hash of block 0.
Any suggestions? 
Maybe just add verification code that applies to block 0?

Reply:
The hash pointer on block 1 will be validated, though, since in the code that calls the puller, we compute the actual header hash of the previous block (0), and compare it to the expected hash in the header of the current block (1).

Reply:
Nevermind, forget it. I thought that we can skip signature validation for genesis in another way.